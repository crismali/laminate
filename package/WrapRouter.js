import { Component } from 'react'
import PropTypes from 'prop-types'

export class WrapRouter extends Component {
  static propTypes = {
    children: PropTypes.any.isRequired,
    router: PropTypes.object.isRequired
  }

  static childContextTypes = {
    router: PropTypes.object.isRequired
  }

  static contextTypes = {
    router: PropTypes.object.isRequired
  }

  getChildContext () {
    return {
      router: {
        ...this.context.router,
        ...this.props.router
      }
    }
  }

  render () {
    return this.props.children
  }
}
