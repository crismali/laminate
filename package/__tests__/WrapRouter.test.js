/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import { WrapRouter } from '../WrapRouter'

it('renders its children', () => {
  const Child = () => 'a child'
  const subject = shallow(<WrapRouter router={{}}>
    <Child />
  </WrapRouter>, { context: { router: {} } })
  expect(subject).toMatchSnapshot()
})

it('overrides the router in context', () => {
  const Child = () => 'a child'
  const router = { a: 'router', with: 'stuff' }
  const newRouter = { my: 'router', with: 'my stuff' }
  const subject = shallow(<WrapRouter router={newRouter}>
    <Child />
  </WrapRouter>, { context: { router } })
  expect(subject.instance().getChildContext()).toEqual({
    router: {
      a: 'router',
      my: 'router',
      with: 'my stuff'
    }
  })
})
