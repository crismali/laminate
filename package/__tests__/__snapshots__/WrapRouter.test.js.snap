// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders its children 1`] = `<Child />`;
